File: workspace/java/origin/playground-05-oop/src/main/java/org/example/enums/Gender.java
/*****************************************************
 * @software: IntelliJ IDEA
 * @author: Lionel Johnson
 * @contact: https://fairy.host
 * @organization: https://github.com/FairylandFuture
 * @datetime: 2025-08-16 13:33:47 UTC+08:00
 ****************************************************/
package org.example.enums;

import jakarta.validation.constraints.NotNull;

/**
 * @author Lionel Johnson
 */
public enum Gender {
    MALE("男"),
    FEMALE("女");
    
    private final String chineseName;
    
    Gender(String chineseName) {
        this.chineseName = chineseName;
    }
    
    public static Gender fromString(@NotNull String name) {
        String input = name.trim().toUpperCase();
        for (Gender color : values()) {
            if (color.chineseName.equals(name) || color.name().equals(input)) {
                return color;
            }
        }
        throw new IllegalArgumentException("无效枚举: " + name);
    }
    
    @Override
    public String toString() {
        return chineseName;
    }
}

File: workspace/python/expert/agent-dev/langchain-rag-lab/common/enum/gender.py
# coding: UTF-8
"""
@software: PyCharm
@author: Beau Dean
@contact: https://fairy.host
@organization: https://github.com/FairylandFuture
@datetime: 2026-04-24 13:09:56 UTC+08:00
"""
from __future__ import annotations

import typing as t

from enum import Enum


class GenderEnum(Enum):
    MALE = "男"
    FEMALE = "女"

    @property
    def value(self) -> str:
        return t.cast(str, t.cast(object, super().value))
